util/src/main/java/org/killbill/billing/util/collect/SimpleHashBiMap.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2020-2022 Equinix, Inc
+ * Copyright 2014-2022 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.util.collect;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+/**
+ * Simple {@link BiMap} implementation: This class extends {@link HashMap} and use most of its functionality except
+ * for method that described in {@link BiMap}.
+ */
+public class SimpleHashBiMap<K, V> extends HashMap<K, V> implements BiMap<K, V> {
+
+    static final Set<Collector.Characteristics> CH_ID = Collections.unmodifiableSet(EnumSet.of(Collector.Characteristics.IDENTITY_FINISH));
+
+    public V put(final K key, final V value) {
+        if (values().contains(value)) {
+            throw new IllegalArgumentException("Value already present: " + value);
+        }
+        return super.put(key, value);
+    }
+
+    private K getKeyByValue(final V value) {
+        return entrySet().stream()
+                .filter(entry -> value.equals(entry.getValue()))
+                .findFirst().get().getKey();
+    }
+
+    @Override
+    public V forcePut(final K key, final V value) {
+        if (values().contains(value)) {
+            final K k = getKeyByValue(value);
+            final boolean isRemoved = remove(k, value);
+            if (!isRemoved) {
+                throw new IllegalStateException("Illegal BiMap#focePut() behaviour. Remove operation failed");
+            }
+        }
+        return super.put(key, value);
+    }
+
+    @Override
+    public BiMap<V, K> inverse() {
+        return this.keySet()
```

Comment:
It looks like we're only using BiMap in tests, so this is probably fine, but I suspect the Guava version is O(1) while this is O(n). Might be worth adding a comment in case we decide to use this for production code at some point.